### ISSUE TITLE
getCurrentPosition throws CODE 1 error on iOS

### ISSUE THREAD

**Issue by pabichw** [react-native-geolocation/react-native-geolocation#139]
When location request dialog pops up `getCurrentPosition` function throws following error almost immediately (after 2-3):
`
{
  code: 1,
  message: "User denied access"
}
`
Code:
`Geolocation.getCurrentPosition(
      (pos) => {
        console.log('got position', pos);
        debugger;
      },
      (err) => {
        console.error('geo Error', err);
        debugger;
      },
      { enableHighAccuracy: false, timeout: 20000, maximumAge: 1000 },
    );
`

**Comment by pku-arun** [comment]
I am facing the same issue. Is there any fix for this?

**Comment by infazA** [comment]
Did anyone find a solution for this issue ? Im also facing the same.

**Comment by Leehee1on** [comment]
Same issue for me, but change library
https://github.com/Agontuk/react-native-geolocation-service

// function 
Geolocation.requestAuthorization('whenInUse')
        .then((result) => {
          if (result === 'granted') {
         // success case
          }
        })
        .catch((error) => console.log(error));

Is my solution

**Comment by 0x1bitcrack3r** [comment]
Any other workaround instead of install new npm package?

**Comment by 0x1bitcrack3r** [comment]
@pabichw try reducing timeout value in this
`{ enableHighAccuracy: false, timeout: 20000, maximumAge: 1000 }` this solved my problem

**Comment by infazA** [comment]
@pabichw Were you able to find a solution for this ?

**Comment by pabichw** [comment]
@infazA No, it didn’t happen in my team mate’s builds so I decided to postpone this

**Comment by nica0012** [comment]
This also happens to me too. Super weird because like you said it only happens if you don't allow location right away... After waiting a couple seconds it will give you that error...? Any solutions?

**Comment by kartavyaparekh96** [comment]
This also happens to me too.

**Comment by nica0012** [comment]
This is becoming a big problem for users. Has anyone resolved this?

**Comment by nica0012** [comment]
A potential fix... I re-ran the function on the catch, and well - it worked

**Comment by 0x1bitcrack3r** [comment]
> A potential fix... I re-ran the function on the catch, and well - it worked

How will you handle the error then. What if it fails again in the catch block. I don't think this a potential fix. Did you try increasing the timeout value `{ enableHighAccuracy: false, timeout: 20000, maximumAge: 1000 },` this solved my problem. I was facing same issue in iOS 14

**Comment by nica0012** [comment]
> > A potential fix... I re-ran the function on the catch, and well - it worked
> 
> 
> 
> How will you handle the error then. What if it fails again in the catch block. I don't think this is a potential fix. Did you try increasing the timeout value `{ enableHighAccuracy: false, timeout: 20000, maximumAge: 1000 },` this solved my problem. I was facing the same issue in iOS 14

I did try this with a value of 200000, but still no luck. Whenever this was happening and I did a fast refresh it would work which is why I tried my solution. I don't even want to call it a solution because it's so quick and dirty but it's a hotfix (for now) to resolve this crash that's happening for many people. This package has not been updated in a while I'm wondering if it had anything to do with new IOS versions.

**Comment by nica0012** [comment]
> Same issue for me, but change library
> https://github.com/Agontuk/react-native-geolocation-service
> 
> // function
> Geolocation.requestAuthorization('whenInUse')
> .then((result) => {
> if (result === 'granted') {
> // success case
> }
> })
> .catch((error) => console.log(error));
> 
> Is my solution

I ended up going this route, works like a charm.

**Comment by yasinugrl** [comment]
Any solition?

**Comment by alicancelik** [comment]
This also happens to me too.

**Comment by AlexeyNovikov** [comment]
Hi guys. I've created PR for this issue. https://github.com/react-native-geolocation/react-native-geolocation/pull/168
This happens on iOS 14+. Feel free to check it out and ping me here if you find any bugs.

**Comment by asafkorem** [comment]
Any update here?

**Comment by ismailharmanda** [comment]
Any solution?

**Comment by Mohamed-akmal0** [comment]
any solution to this still getting the same issue on simulator and real device

react native version => 0.71.2
